lib/bap_disasm/bap_disasm_rec.ml

```diff
@@ -213,26 +203,75 @@ let next dis s =
 
 let stop_on = [`Valid]
 
-(* find intersections, i.e. instructions which share bytes, but have different address,
-   and update destinations for those of them, which also share max address to
-   split a block in such place *)
+module Dest = struct
+  let rec compare_edge e e' =
+    if e = e' then 0
+    else
+      match e, e' with
+      | `Fall, _ -> -1
+      | `Jump, _ -> 1
+      | _ -> - (compare_edge e' e)
+
+  let compare (d,e) (d',e') =
+    match d, d' with
+    | None,None -> compare_edge e e'
+    | Some _, None -> 1
+    | None, Some _ -> -1
+    | Some d, Some d' ->
+       let x = Word.compare d d' in
+       if x = 0 then compare_edge e e'
+       else x
+
+  module Set = Set.Make(struct
+      type t = Brancher.dest [@@deriving sexp]
+      let compare = compare
+    end)
+
+end
+
+
+let equal_dest (d,e) (d',e') =
```

Comment:
You already have `compare_dest` function, so `equal_dest` should be `compare_dest x y = 0`, or just use the derived equality from the `Base.Comparable` interface. 

If your `equal x y` is not defined as `compare x y = 0` mathematically, then this is just a wrong way of doing things.